services/alertmanager/alert.go

```diff
@@ -0,0 +1,137 @@
+// pmm-managed
+// Copyright (C) 2017 Percona LLC
+//
+// This program is free software: you can redistribute it and/or modify
+// it under the terms of the GNU Affero General Public License as published by
+// the Free Software Foundation, either version 3 of the License, or
+// (at your option) any later version.
+//
+// This program is distributed in the hope that it will be useful,
+// but WITHOUT ANY WARRANTY; without even the implied warranty of
+// MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE. See the
+// GNU Affero General Public License for more details.
+//
+// You should have received a copy of the GNU Affero General Public License
+// along with this program. If not, see <https://www.gnu.org/licenses/>.
+
+package alertmanager
+
+import (
+	"fmt"
+
+	"github.com/percona/pmm/api/alertmanager/ammodels"
+	"github.com/pkg/errors"
+	"github.com/prometheus/common/model"
+
+	"github.com/percona/pmm-managed/models"
+)
+
+// Severity defines alert severity.
+type Severity string
+
+// severities
+const (
+	Error   = Severity("error")
+	Warning = Severity("warning")
+	Info    = Severity("info")
+)
+
+// AlertParams defines alert parameters.
+type AlertParams struct {
+	Name        string
+	Summary     string
+	Description string
+	Severity    Severity
+
+	Node    *models.Node
+	Service *models.Service
+	Agent   *models.Agent
+}
+
+// validate checks parameters and fills defaults.
+func (ap *AlertParams) validate() error {
+	if ap.Name == "" {
+		return errors.New("empty Name")
+	}
+	if ap.Summary == "" {
+		return errors.New("empty Summary")
+	}
+	if ap.Description == "" {
+		return errors.New("empty Description")
+	}
+
+	if ap.Severity == "" {
+		ap.Severity = Info
+	}
+
+	return nil
+}
+
+// makeAlert makes alert from given parameters.
+func makeAlert(params *AlertParams) (*ammodels.PostableAlert, error) {
+	if err := params.validate(); err != nil {
+		return nil, err
+	}
+
+	labels, err := models.MergeLabels(params.Node, params.Service, params.Agent)
+	if err != nil {
+		return nil, err
+	}
+
+	labels[model.AlertNameLabel] = params.Name
+	labels["severity"] = string(params.Severity)
+	labels["stt_check"] = "1"
+
+	return &ammodels.PostableAlert{
```

Comment:
The whole block file is not used in this PR. The only reason it is there is to avoid future merge conflicts. That code will be improved, but that's not important now.